storage/storage_dest.go

```diff
@@ -313,13 +313,20 @@ func (f *zstdFetcher) GetBlobAt(chunks []chunked.ImageSourceChunk) (chan io.Read
 // It is available only if SupportsPutBlobPartial().
 // Even if SupportsPutBlobPartial() returns true, the call can fail, in which case the caller
 // should fall back to PutBlobWithOptions.
-func (s *storageImageDestination) PutBlobPartial(ctx context.Context, chunkAccessor private.BlobChunkAccessor, srcInfo types.BlobInfo, options private.PutBlobPartialOptions) (private.UploadedBlob, error) {
+func (s *storageImageDestination) PutBlobPartial(ctx context.Context, chunkAccessor private.BlobChunkAccessor, srcInfo types.BlobInfo, options private.PutBlobPartialOptions) (_ private.UploadedBlob, Err error) {
 	fetcher := zstdFetcher{
 		chunkAccessor: chunkAccessor,
 		ctx:           ctx,
 		blobInfo:      srcInfo,
 	}
 
+	var perr chunked.ErrFallbackToOrdinaryLayerDownload
+	defer func() {
```

Comment:
Should this be checked _only_ for the `GetDiffer` call? That’s the only case where it can currently be returned.

Or maybe also `PrepareStagedLayer` — but in the `out.CompressedDigest != blobDigest` case I’d prefer to hard-fail and not fall back.

Reply:
in this case we should be already safe no?  The error we return is not wrapped by `chunked.ErrFallbackToOrdinaryLayerDownload`:

```go
		if out.CompressedDigest != blobDigest {
			return private.UploadedBlob{}, fmt.Errorf("internal error: PrepareStagedLayer returned CompressedDigest %q not matching expected %q",
				out.CompressedDigest, blobDigest)
		}
```

Reply:
My mistake, you are absolutely correct.